core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2108,12 +2108,18 @@ public boolean enableDeduplicationWithUpsertKeys(LongList columnsIndexes) {
         }
 
         columnsIndexes.sort();
-        for (int i = 0, j = 0; i < columnCount && j < columnsIndexes.size(); i++) {
-            if (i == columnsIndexes.getQuick(j)) {
+        int j = 0;
+        long nextDedupColIndex = columnsIndexes.size() > j ? (int) columnsIndexes.getQuick(j) : Long.MAX_VALUE;
+        for (int i = 0; i < columnCount; i++) {
+            if (i == nextDedupColIndex) {
```

Comment:
this logic looks a little messy. the `nextDedupColIndex` is long and can be Long.MAX_VALUE. Why cant this operate in one dimension (int) instead of two (int,long)